tests/build-script.rs

```diff
@@ -256,20 +257,18 @@ fn links_duplicates() {
     assert_that(p.cargo("build"),
                 execs().with_status(101)
                        .with_stderr("\
-[ERROR] multiple packages link to native library `a`, but a native library can \
-be linked only once
-
-package `foo v0.5.0 ([..])`
-links to native library `a`
-
-package `a-sys v0.5.0 ([..])`
-    ... which is depended on by `foo v0.5.0 ([..])`
-also links to native library `a`
+error: failed to select a version for `a-sys`.
+all possible versions conflict with previously selected packages.
+required by package `foo v0.5.0 ([..])`
+  multiple packages link to native library `a`, but a native library can be linked only once.
+  previously selected package `foo v0.5.0 ([..])`
+  possible versions to select: 0.5.0
```

Comment:
I wonder if we could perhaps tidy up the wording around this sort of error? I found this sort of hard to follow as there's a lot of mentions of versions and such. Maybe something like:

```
error: failed to select a version for `a-sys`
  ... required by `foo v0.5.0`

the package `a-sys` links to the native library `a`, but it conflicts
with a previous package which links to `a` as well:

  foo v0.5.0
     .. required by ...

failed to select a version for `a-sys` which could resolve this conflict
```

I do realize that most of this is fitting in the mold of the previous error messages, but perhaps we can just tidy up the wording around them?

Reply:
I will see what I can do. :-)

One thing to keep in mind is that different versions of a-sys may have different errors we need to report. For example only some versions have the link attribute. So we may, at least in theory, have to coherently report on a link error with 'a-fork-sys' and a semver error with 'a-sys' from before it had a link attribute.